src/main/java/org/chrisle/netbeans/plugins/nbscratchfile/CreateScratchFile.java

```diff
@@ -1,158 +1,142 @@
-                focusOwner = focusOwner.getParent();
-            }
-        }
-
-        dialog.setLocationRelativeTo(focusOwner);
-        dialog.setVisible(true);
-    }
+package org.chrisle.netbeans.plugins.nbscratchfile;
+
+import java.awt.Component;
+import java.awt.KeyboardFocusManager;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.KeyEvent;
+import java.awt.event.WindowEvent;
+import java.awt.event.WindowFocusListener;
+import javax.swing.JComponent;
+import javax.swing.JDialog;
+import javax.swing.KeyStroke;
+import net.java.html.js.JavaScriptBody;
+import org.netbeans.api.htmlui.HTMLComponent;
+import org.netbeans.api.htmlui.HTMLDialog;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.awt.ActionReferences;
+import org.openide.awt.ActionRegistration;
+import org.openide.util.NbBundle.Messages;
+
+@ActionID(
+        category = "Tools",
+        id = "org.chrisle.netbeans.plugins.nbscratchfile.CreateScratchFile"
+)
+@ActionRegistration(
+        displayName = "#CTL_CreateScratchFile",
+        iconBase = "org/chrisle/netbeans/plugins/nbscratchfile/add_file.png"
+)
+@ActionReferences({
+    @ActionReference(path = "Menu/File", position = 150)
+    ,
+    @ActionReference(path = "Shortcuts", name = "DOS-N")
+})
+@Messages("CTL_CreateScratchFile=New Scratch File...")
+public final class CreateScratchFile implements ActionListener {
+    private final JDialog dialog;
+    private final JComponent jfxPanel;
+    private final NbScratchFileViewModel viewModel;
+
+    public CreateScratchFile() {
+        dialog = new JDialog();
+        viewModel = new NbScratchFileViewModel(dialog);
+        jfxPanel = Pages.initWebView(viewModel);
+        initDialog();
+    }
+
+    private void initDialog() {
+        dialog.add(jfxPanel);
+        dialog.setSize(700, 450);
+        dialog.setResizable(false);
+        dialog.setAlwaysOnTop(true);
+        dialog.setUndecorated(true);
+        dialog.addWindowFocusListener(new WindowFocusListener() {
+            @Override
+            public void windowGainedFocus(WindowEvent e) {
+            }
+
+            @Override
+            public void windowLostFocus(WindowEvent e) {
+                dialog.setVisible(false);
+            }
+        });
+
+        dialog.getRootPane().registerKeyboardAction((ActionEvent e1) -> {
+            dialog.setVisible(false);
+        }, KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0), JComponent.WHEN_IN_FOCUSED_WINDOW);
+    }
+
+    @JavaScriptBody(
+        args = { "id", "tag", "css" }, body = "\n"
+                + "var sourceElem = id ? document.getElementById(id) : document.getElementsByTagName(tag)[0];\n"
+                + "sourceElem.setAttribute('style', css);\n"
+                + ""
+    )
+    private static native void colorizeElement(String id, String tag, String css);
+    
+    /* this would rather be done on the TypeScript side as it doesn't talk to Java:
```

Comment:
There doesn't seem to be any need to do this on the Java side as there is no Java interaction needed. I can do it, if you want however.